File: frontend/src/types/asset.ts
/** Character asset — an anime character with reference image */
export interface Character {
  id: string
  project_id: string
  name: string
  description: string
  appearance: string
  personality: string
  image_url: string | null
  thumbnail_url: string | null
  reference_prompt: string
  voice_profile_id: string | null
  status: 'pending' | 'generating' | 'done' | 'failed'
  created_at: string
  updated_at: string
}

/** Scene / background asset */
export interface Scene {
  id: string
  project_id: string
  name: string
  description: string
  setting: string
  time_of_day: string
  weather: string
  image_url: string | null
  thumbnail_url: string | null
  reference_prompt: string
  status: 'pending' | 'generating' | 'done' | 'failed'
  created_at: string
  updated_at: string
}

/** Keyframe generated for a shot */
export interface KeyFrame {
  id: string
  shot_id: string
  image_url: string
  thumbnail_url: string
  prompt: string
  width: number
  height: number
  created_at: string
}

/** Animation clip generated from a keyframe */
export interface Animation {
  id: string
  keyframe_id: string
  video_url: string
  duration: number
  fps: number
  created_at: string
}

export type AssetStatus = 'pending' | 'generating' | 'done' | 'failed'

File: backend/app/schemas/asset.py
"""Asset Pydantic schemas."""
from datetime import datetime
from typing import Optional
from pydantic import BaseModel


class CharacterResponse(BaseModel):
    id: str
    project_id: str
    name: str
    description: str
    appearance: str
    personality: str
    image_url: Optional[str]
    thumbnail_url: Optional[str]
    reference_prompt: str
    voice_profile_id: Optional[str] = None
    status: str
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class SceneResponse(BaseModel):
    id: str
    project_id: str
    name: str
    description: str
    setting: str
    time_of_day: str
    weather: str
    image_url: Optional[str]
    thumbnail_url: Optional[str]
    reference_prompt: str
    status: str
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
